lib/ddtrace/contrib/racecar/patcher.rb

```diff
@@ -0,0 +1,78 @@
+require 'ddtrace/ext/app_types'
+
+module Datadog
+  module Contrib
+    module Racecar
+      # Provides instrumentation for `racecar` through ActiveSupport instrumentation signals
+      module Patcher
+        include Base
+        NAME = 'racecar.consumer'.freeze
+        register_as :racecar
+        option :tracer, default: Datadog.tracer
+        option :service_name, default: 'racecar'
+
+        class << self
+          def patch
+            return patched? if patched? || !compatible?
+
+            ::ActiveSupport::Notifications.subscribe('start_process_batch.racecar', &method(:start))
+            ::ActiveSupport::Notifications.subscribe('start_process_message.racecar', &method(:start))
+            ::ActiveSupport::Notifications.subscribe('process_batch.racecar', &method(:finish))
+            ::ActiveSupport::Notifications.subscribe('process_message.racecar', &method(:finish))
```

Comment:
I think you need to separate message processing from batch processing – at least give the spans different names.

Reply:
Hey @dasch! Apologies for the late reply. I'm not sure if I fully understand the problem here. As far as I understand each worker either implements a `process` or `process_batch` method, correct? So each span will belong to a resource that univocally represents either batch or a message worker.

Reply:
That's true, but a service can change from one to the other, and often will – batching is a performance optimization. Naming the top-level span either `process_message` or `process_batch` would make it clear what the trace represents.

Reply:
I see! I'll make change that change then!